test/widgets/actions_test.dart

```diff
@@ -267,6 +268,27 @@ void main() {
           });
     });
 
+    testWidgets('MentionsNarrow on legacy server', (WidgetTester tester) async {
+      const narrow = MentionsNarrow();
+      final message = eg.streamMessage(flags: [MessageFlag.mentioned]);
+      final unreadMsgs = eg.unreadMsgs(mentions: [message.id]);
+      await prepare(tester, unreadMsgs: unreadMsgs);
+      connection.zulipFeatureLevel = 154;
+      connection.prepare(json:
+        UpdateMessageFlagsResult(messages: [message.id]).toJson());
+      markNarrowAsRead(context, narrow, true);  // TODO move legacy-server check inside markNarrowAsRead
+      await tester.pump(Duration.zero);
+      check(connection.lastRequest).isA<http.Request>()
+        ..method.equals('POST')
+        ..url.path.equals('/api/v1/messages/flags')
+        ..bodyFields.deepEquals({
+            'messages': jsonEncode([message.id]),
+            'op': 'add',
+            'flag': 'read',
+          });
+        await tester.pumpAndSettle();
```

Comment:
nit: overindented; but also do we need this line? the similar test just above doesn't seem to have it